src/Shared/ProblemDetails/ProblemDetailsJsonConverter.cs

```diff
@@ -8,15 +8,18 @@
 
 namespace Microsoft.AspNetCore.Http;
 
-internal sealed class ProblemDetailsJsonConverter : JsonConverter<ProblemDetails>
+internal sealed partial class ProblemDetailsJsonConverter : JsonConverter<ProblemDetails>
 {
     private static readonly JsonEncodedText Type = JsonEncodedText.Encode("type");
     private static readonly JsonEncodedText Title = JsonEncodedText.Encode("title");
     private static readonly JsonEncodedText Status = JsonEncodedText.Encode("status");
     private static readonly JsonEncodedText Detail = JsonEncodedText.Encode("detail");
     private static readonly JsonEncodedText Instance = JsonEncodedText.Encode("instance");
 
-    [UnconditionalSuppressMessage("Trimmer", "IL2026", Justification = "Trimmer does not allow annotating overriden methods with annotations different from the ones in base type.")]
+    [RequiresUnreferencedCode("JSON serialization and deserialization of ProblemDetails.Extensions might require types that cannot be statically analyzed.")]
+    [RequiresDynamicCode("JSON serialization and deserialization of ProblemDetails.Extensions might require types that cannot be statically analyzed.")]
+    [UnconditionalSuppressMessage("Trimming", "IL2046:'RequiresUnreferencedCodeAttribute' annotations must match across all interface implementations or overrides.", Justification = "<Pending>")]
+    [UnconditionalSuppressMessage("AOT", "IL3051:'RequiresDynamicCodeAttribute' annotations must match across all interface implementations or overrides.", Justification = "<Pending>")]
```

Comment:
Normally, the way we handle this is we mark the constructor of the type, or the whole type itself, as `RequiresUnreferencedCode`/`RequiresDynamicCode`. That says "creating one of these is not compatible", and moves the warning to the caller that instantiates it.

Reply:
The problem with adding the attributes to the entire type is that the converter type is then used in System.Text.Json serialization generated code. There is no chance to suppress it there.

Reply:
Suppressing the warning is not correct then, if we can't guarantee this code works with trimming and AOT. We will need to find another way designing this code to work with these form factors.

Can you open an issue for this, and add it as a comment here, so we can move forward with the rest of the changes?

Reply:
Working on this in https://github.com/dotnet/aspnetcore/pull/45646